lib/login/google_Sign_In.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class UserAuth {
+  UserAuth({
+    @required this.googleSignIn,
+    @required this.firebaseAuth,
+  });
+
+  GoogleSignIn googleSignIn;
+  FirebaseAuth firebaseAuth;
+
+  Future<FirebaseUser> signInWithGoogle() async {
+    FirebaseUser user;
+    final bool isSignedIn = await googleSignIn.isSignedIn();
+    if (isSignedIn) {
+      user = await firebaseAuth.currentUser();
+    } else {
+      final GoogleSignInAccount googleUser = await googleSignIn.signIn();
```

Comment:
wrap everything in this else statement in a function called something like authenticateUser()